src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PhysicalFilesWatcher.cs

```diff
@@ -130,7 +130,10 @@ public IChangeToken CreateFileChangeToken(string filter)
             }
 
             IChangeToken changeToken = GetOrAddChangeToken(filter);
-            TryEnableFileSystemWatcher();
+            if (!PollForChanges)
```

Comment:
It might be clearer if we just made the constructor ignore the FileSystemWatcher entirely.  Moreover, I wonder if we should be adding a constructor overload which doesn't require a `FileSystemWatcher`.

Reply:
Per offline conversation to make this less risky we'll make the constructor allow for null FSW.  In the case of a null FSW and PollForChanges=true we'll permit only polling mode.  We'll make PFP construct the type this way when UsePolling is true.